js/src/pages/onboarding/index.js

```diff
@@ -12,6 +14,15 @@ import SetupStepper from './setup-stepper';
  */
 const Onboarding = () => {
 	useLayout( 'full-page' );
+	useUpdateRestAPIAuthorizeStatusByUrlQuery();
+
+	const canContinueRendering = useAutoWPComAppAuthorization();
+
+	// Render nothing as the requirement is to make it look like the redirections
+	// between Google authorization and WPCOM app authorization are seamless.
+	if ( ! canContinueRendering ) {
+		return null;
```

Comment:
Not sure if we could add a spinner here since the blank page before redirecting to WPCOM feels a bit long in my site.

Reply:
Thanks for the suggestion!

I also needed to wait for a while during this process. For a moment I would suspect the blank page was stuck or not responding. Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/2802/commits/77e5beb6ae5d951fb384f5e0b666415bbd551fa0

https://github.com/user-attachments/assets/cc43fc15-7119-4f14-b01b-1b3582735947